### DISCUSSION TITLE
Having difficulty understanding the grblHAL documentation

### DISCUSSION THREAD

**Discussion by howardlowndes** [grblHAL/core#213] [Q&A]
My background is in IT mostly at levels above C++ and with Linux and more recently with Arduino.
I am currently studying g code and its use in CNC machines and that has been around Arduino Nano and Grbl. Now I want to move forward from there and look at using grblHAL on the Raspberry RP2040 and the Pico BOB.
This is where I am getting confused.
I have pulled the grblHAL/core and the grblHAL/RP2040 and I want to use these in conjunction with the Arduino IDE - either 1.8.x or 2.0.x.
The documentation states that the **grblHAL-xxx** zip file can be installed in the Arduino libraries as either a ZIP file or as a folder. However, pulling from the git repository provides **core-master.zip** not anything resembling **grblHAL-xxx**. I can install that but it appears as **core** in the libraries. The documentation says that **grblHAL-xxx** can also be installed even if it not a ZIP file. THat just doesn't work.

Moving on to the grblHAL/RP2040 clone; how is that associated with grblHAL/core? Do I need both clones? How are they integrated, or does grblHAL/RP2040 include grblHAL/core?

I apologise if my ignorance is frustrating, but I really am having difficulty following the thread of the documentation.

**Comment by dresco** [comment]
Hi, I suspect the documentation page you are looking at just doesn't cover the RP2040 build (yet).

The RP2040 driver is not built with Arduino IDE. The [README](https://github.com/grblHAL/RP2040/blob/master/README.md) for the RP2040 repo links to a PDF with instructions for installing the SDK, and integrating with VS Code.

Your starting point would be to clone the [RP2040](https://github.com/grblHAL/RP2040) driver repo   (making sure to include submodules, as these are how the grblHAL core files and all the plugins are delivered) `git clone https://github.com/grblHAL/RP2040.git --recurse-submodules`

**Comment by terjeio** [comment]
The new [Web Builder](https://github.com/grblHAL/core/discussions/203) now has experimental support for building the RP2040 driver. Availability may be a bit erratic for a while as the backend service is work in progress.

Flashing the Pico is easy, plug in the Pico while pressing the Boot button. This will mount a drive where the executable (_firmware.uf2_) can be dropped.
